### ISSUE TITLE
Support for multi tenancy column discriminator 

### ISSUE THREAD

**Issue by guitcastro** [quarkusio/quarkus#28644]
### Description

[#5681](https://github.com/quarkusio/quarkus/issues/5681) added support to hibernate, however the column discriminator was not supported, because by the time, Hibernate itself didn't have fully support for it.
```

        if (multiTenancyStrategy == MultiTenancyStrategy.DISCRIMINATOR) {
            // See https://hibernate.atlassian.net/browse/HHH-6054
            throw new ConfigurationException("The Hibernate ORM multitenancy strategy "
                    + MultiTenancyStrategy.DISCRIMINATOR + " is currently not supported");
        }
```

However, [HHH-6054](https://hibernate.atlassian.net/browse/HHH-6054) is now completed and we can have support for discriminator column.

**Comment by guitcastro** [comment]
Linking [26986](https://github.com/quarkusio/quarkus/issues/26986) for reference.

**Comment by humcqc** [comment]
Linking 32686 for reference, WIP.

**Comment by yrodiere** [comment]
Fixed by #32855
